packages/core/src/lib/connection_manager.ts

```diff
@@ -60,6 +65,27 @@ export class ConnectionManager {
     this.run()
       .then(() => log(`Connection Manager is now running`))
       .catch((error) => log(`Unexpected error while running service`, error));
+
+    // libp2p emits `peer:discovery` events during its initialisation
+    // which means that before the ConnectionManager is initalised, some peers may have been discovered
```

Comment:
shouldn't we initialize connection manager before starting the libp2p node? wouldn't that solve the issue with this race condition?